HPXMLtoOpenStudio/measure.rb

```diff
@@ -1853,9 +1853,24 @@ def self.apply_shading(model, window_or_skylight, shading_polygon, parent_surfac
       shading_surface.setTransmittanceSchedule(shading_schedules[trans_values].schedule)
 
       # Adjustment to default view factor is used to reduce ground diffuse solar
-      avg_trans_value = trans_values.sum(0.0) / 12.0 # FUTURE: Create EnergyPlus actuator to adjust this
-      default_vf_to_ground = ((1.0 - Math::cos(parent_surface.tilt)) / 2.0).round(2)
-      sub_surface.setViewFactortoGround(default_vf_to_ground * avg_trans_value)
+      actuator = OpenStudio::Model::EnergyManagementSystemActuator.new(sub_surface, *EPlus::EMSActuatorSurfaceViewFactorToGround)
+      actuator.setName("#{sub_surface.name.to_s.gsub(' ', '_').gsub('-', '_')}_actuator")
+
+      # EMS to actuate schedule
+      program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
+      program.setName("#{sub_surface.name.to_s.gsub('-', '_')} View Factor To Ground Program")
+      program.addLine("IF (DayOfYear >= 1) && (DayOfYear <= 120)") # FIXME: Is there a way to parameterize DayOfYear so that I can avoid hard-coding it?
```

Comment:
@shorowit This EMSProgram is to overwrite the `view factor to ground` of the subsurfaces. I've hardcoded `DayOfYear` for the first pass but I'd like to avoid it. Any ideas on how to parameterize `DayOfYear`? 

If we determined to parameterize `DayOfYear`, we probably need some logic to find the nearest day that’s a multiple of 20 in order to synchronize with the default period frequency (i.e., 20 days) of the detailed diffuse algorithm.

Reply:
I suggest using the schedule value from the `trans_sch` in your EMS program instead, using an EMS sensor. The schedule already incorporates the summer vs winter seasons. If you search the code, you'll find many examples of EMS programs that use sensors of schedules.